### PR TITLE
Created Registration test module and moved relevant tests from host.py

### PR DESCRIPTION
**Pull request by shweta83**
### Problem Statement
Test cases for registration component were added in ui/test_host.py and it was difficult to track the tests under that component.

### Solution
Created a new test module  tests/foreman/ui/test_registration.py so that the relevant tests can be tracked under correct component.

### CODE REVIEW

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_registration.py

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_registration.py

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_registration.py -k test_positive_host_registration_with_non_admin_user

File: tests/foreman/ui/test_host.py [hunk 1]
```diff
@@ -1027,289 +1023,6 @@ def test_positive_validate_inherited_cv_lce_ansiblerole(session, target_sat, mod
...
-            },
-            full_read=True,
-        )
-        assert new_org.name in cmd['general']['organization']
-        assert new_ak.name in cmd['general']['activation_keys']
-
-
 @pytest.mark.tier2
 def test_global_registration_with_capsule_host(
     session,
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_host.py:1028]: Could you also move this test (or look for other tests) along with other global_registration tests?

Comment (shweta83) [reply tests/foreman/ui/test_host.py:1028]: I feel this is more host/capsule related test than registration test.

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_registration.py

File: tests/foreman/ui/test_host.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/ui/test_host.py:1028]: yes, this is definately a registration test against capsule

File: tests/foreman/ui/test_registration.py [hunk 2]
```diff
@@ -121,3 +128,496 @@ def test_negative_global_registration_without_ak(
...
+    """
+    client = rhel8_contenthost
+    repo_name = 'foreman_register'
+    repo_url = settings.repos.gr_yum_repo.url
+    repo_gpg_url = settings.repos.gr_yum_repo.gpg_url
+    with session:
+        cmd = session.host.get_register_command(
+            {
+                'general.operating_system': default_os.title,
+                'general.capsule': default_smart_proxy.name,
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_registration.py:389]:
```suggestion
```

File: tests/foreman/ui/test_registration.py [hunk 3]
```diff
@@ -121,3 +128,496 @@ def test_negative_global_registration_without_ak(
...
+        5. re-register the same host again and check it is getting registered
+
+    :parametrized: yes
+    """
+    client = rhel7_contenthost
+    with session:
+        cmd = session.host.get_register_command(
+            {
+                'general.operating_system': default_os.title,
+                'general.capsule': default_smart_proxy.name,
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_registration.py:449]:
```suggestion
```

File: tests/foreman/ui/test_registration.py [hunk 4]
```diff
@@ -121,3 +128,496 @@ def test_negative_global_registration_without_ak(
...
+        2. use that curl token to execute other api calls e.g. GET /hosts, /users
+
+    :parametrized: yes
+    """
+    client = rhel8_contenthost
+    with session:
+        cmd = session.host.get_register_command(
+            {
+                'general.operating_system': default_os.title,
+                'general.capsule': default_smart_proxy.name,
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_registration.py:491]:
```suggestion
```

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_registration.py

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_registration.py

File: tests/foreman/ui/test_registration.py (hunk 4 above)
Comment (shweta83) [reply tests/foreman/ui/test_registration.py:471]: I am going to address all the updates in test in https://github.com/SatelliteQE/robottelo/pull/14001

File: tests/foreman/ui/test_registration.py (hunk 3 above)
Comment (shweta83) [reply tests/foreman/ui/test_registration.py:431]: I am going to address all the updates in test in https://github.com/SatelliteQE/robottelo/pull/14001

File: tests/foreman/ui/test_registration.py (hunk 2 above)
Comment (shweta83) [reply tests/foreman/ui/test_registration.py:373]: All the updates in test case will be addressed in  https://github.com/SatelliteQE/robottelo/pull/14001

File: tests/foreman/ui/test_host.py (hunk 1 above)
Comment (shweta83) [reply tests/foreman/ui/test_host.py]: Moved this to registration module.

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**